src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3643,7 +3672,7 @@ private static void AddProviderEnumKind(ManifestBuilder manifest, FieldInfo stat
             if (eventData == null || eventData.Length <= eventAttribute.EventId)
             {
                 EventMetadata[] newValues = new EventMetadata[Math.Max(eventData.Length + 16, eventAttribute.EventId + 1)];
-                Array.Copy(eventData, 0, newValues, 0, eventData.Length);
+                Array.Copy(eventData, newValues, eventData.Length);
```

Comment:
@stephentoub Since we have the EventSource code in multiple places it is set up so the files are the same and porting fixes can be done with a merge tool. The usual workflow is to copy CoreCLR (our "master" branch) to the other places we have the code.

Removing the 0s (and changing from nameof above) are both artifacts of this process, since they were added here and not in the master branch. I am neutral about the existence of nameof, if you are arguing for it then I can add them back in, but that causes some extra work in porting changes back to CoreCLR.